### PR TITLE
Add gstream tpc

### PR DESCRIPTION
**Pull request by biozit**

### CODE REVIEW

File: Collectors/DetailedCollector.py [hunk 1]
```diff
@@ -310,12 +340,6 @@ def process_gstream(self, gstream, addr, sid):
                      event["block_bypass_cache"] = event.pop("b_bypass")
                      event["site"] = site
                      event["vo"] = self.returnVO(event["file_path"])
-
-                     if(event["block_hit_cache"] >= event["size"]):
-                         event["cache_request"] = "total"
-                     else:
-                         event["cache_request"] = "partial"
-
```
Comment (jthiltges) [inline Collectors/DetailedCollector.py:318]: Was this removal intended?

File: Collectors/DetailedCollector.py [hunk 2]
```diff
@@ -267,6 +270,33 @@ def returnVO(self,fname):
...
+             event["xeq"]["ip_version"] = event["xeq"].pop("IPv")
+             event["xeq"]["return_code"] = event["xeq"].pop("RC")
+             event["xeq"]["used_streams"] = event["xeq"].pop("Strm")
+             event["xeq"]["flow_direction"] = event["xeq"].pop("Type")
+             event["source"] = event.pop("Src")
+             event["destination"] = event.pop("Dst")
+             event["size"] = event.pop("Size")
+
+             if event["source"].startswith('/store') or event["source"].startswith('/user/dteam'):
+                  self.logger.info("Sending GStream TPC for "+self._wlcg_exchange_tpc)
```
Comment (jthiltges) [inline Collectors/DetailedCollector.py:294]: For log entries on every event, I'd set the level down to debug, and maybe log the contents instead?
```suggestion
                  self.logger.debug("Sending WLCG GStream TPC: %s", str(event))
```

File: Collectors/DetailedCollector.py [hunk 3]
```diff
@@ -267,6 +270,33 @@ def returnVO(self,fname):
...
+                  self.publish("tpc", event, exchange=self._wlcg_exchange_tpc)
+             else:
+                  self.logger.info("Sending GStream TPC for "+self._exchange_tpc)
```
Comment (jthiltges) [inline Collectors/DetailedCollector.py:297]:
```suggestion
                  self.logger.debug("Sending GStream TPC: %s", str(event))
```

**Review by jthiltges — COMMENTED** [review]
Quick comments that may or may not be helpful.
Also, a `.DS_Store` file has sneaked into this PR.

File: Collectors/DetailedCollector.py [hunk 4]
```diff
@@ -267,6 +270,33 @@ def returnVO(self,fname):
...
+
+             self.metrics_q.put({'type': 'gstream_message_tpc', 'count': 1})
+             event["tpc_protocol"] = event.pop("TPC")
+             event["client"] = event.pop("Client")
+             event["xeq"] = event.pop("Xeq")
+
+             self.logger.info(event["xeq"])
+             self.logger.info(type(event["xeq"]))
+             event["xeq"]["begin_transfer"] = event["xeq"].pop("Beg")
+             event["xeq"]["end_trasnfer"] = event["xeq"].pop("End")
```
[the remark this answers is not in the document]
Comment (biozit) [reply Collectors/DetailedCollector.py:284]: thanks!

File: Collectors/DetailedCollector.py (hunk 1 above)
Comment (biozit) [reply Collectors/DetailedCollector.py:318]: Yes, this analysis of the total or partial requests is not accurate as we thought

**Comment by jthiltges** [comment]
LGTM with changes. Look good to squash and merge, @djw8605 ?
